solutions/android/java/com/bitheads/braincloud/AndroidSaveDataHelper.java

```diff
@@ -0,0 +1,70 @@
+
+package com.bitheads.braincloud;
+
+import android.content.Context;
+
+
+public class AndroidSaveDataHelper {
+    private String TAG = AndroidSaveDataHelper;
+    private SharedPreferences sp;
+
+    /**
+     * @param context
+     * @param companyName
+     * @param appName
+     * @param wrapperName
+     * @returns if init success or failed (if Context is null it will return false)
+     */
+
+    public boolean initialize(Context context, String companyName, String appName, String wrapperName) {
+        if (context != null) {
+            if (wrapperName == null) {
+                wrapperName = "";
+            }
+
+            if (companyName == null) {
+                companyName = "";
+            }
+
+            if (appName == null) {
+                appName = "";
+            }
+
+            StringBuilder builder = new StringBuilder();
+            builder.append(wrapperName);
+            builder.append(".");
+            builder.append(companyName);
+            builder.append(".");
+            builder.append("appName");
```

Comment:
passing "appName" not the variable.

Reply:
Is this file necessary in this project? It is not used from AndroidSaveDataHelper.cpp.

Reply:
Also, could this point to this a bug in the java project? Is this file copied from there?